processes/bootstrap/config.go

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2019 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bootstrap
+
+import "github.com/pkg/errors"
+
+// Config represents configuration file for the bootstrap procedure
+type Config struct {
+	// The configuration file consists from a list of processes
+	Processes []Process `json:"processes"`
+}
+
+// Process is a single process instance started by the bootstrap
+type Process struct {
+	// Unique name of the process
+	Name string `json:"name"`
+
+	// File where the log output will be written
+	LogFilePath string `json:"log-file-path"`
+
+	// Path to the process binary file
+	BinaryPath string `json:"binary-path"`
+
+	// A list of arguments the process will be started with
+	Args []string `json:"args"`
+
+	// This parameter allows to define another processes, which will be stopped
+	// when the current instance is terminated (processes are defined by name).
+	TriggerStopFor []string `json:"trigger-stop-for"`
```

Comment:
The `TriggerStopFor` has very narrow use-case and doesn't allow a lot of flexibility for users. It also doesn't fully mirror the functionality provided by `supervisord_kill.py` in vpp-agent (I assume that's why `TriggerStopFor` was introduced here), because this stops other specific process defined for this plugin instead of exiting it. And by adding any other process to the list of processes would break the container-exit functionality (if it wasn't also _chained_ the same way).

To provide more generic behaviour that could be completely flexible and extensible by users I would suggest to remove `TriggerStopFor` field from `Process` and add `Hooks` field of type `[]Hook` to `Config` which could provide arbitrary hooks for this plugin. These hooks could listen to some defined events and execute some external command (script) which would receive attributes of event as environmental variables. This would provide very similar functionality to what supervisord currently provides via [`[eventlistener:x]`](http://supervisord.org/configuration.html#eventlistener-x-section-values) and could be used in any way user wants.

Reply:
> adding any other process to the list of processes would break the container-exit functionality
I see your point, we can just introduce some specific keyword here (or adjust the field) so not all processes need to be listed here.

>  These hooks could listen to some defined events and execute some external command (script) which would receive attributes of an event as environmental variables
I thought we want to get rid of scripts, that's why I found this functionality unnecessary here.

Reply:
We wanted to get rid of python dependency. Simple bash script is harmless.